cogboard-webapp/src/components/MainTemplate/index.js

```diff
@@ -89,16 +92,18 @@ const MainTemplate = () => {
                 <Save />
               </StyledSaveFab>
             )}
-            {isAuthenticated && currentBoardId && (
-              <Fab
-                onClick={handleAddWidgetClick}
-                aria-label="Add Widget"
-                color="primary"
-                data-cy="main-template-add-widget-button"
-              >
-                <Add />
-              </Fab>
-            )}
+            {isAuthenticated &&
+              currentBoardId &&
+              currentBoardType.type !== 'IframeBoard' && (
+                <Fab
+                  onClick={handleAddWidgetClick}
+                  aria-label="Add Widget"
+                  color="primary"
+                  data-cy="main-template-add-widget-button"
+                >
+                  <Add />
+                </Fab>
+              )}
```

Comment:
I wonder what will happen with this logic if we decide to customise appearance of these buttons for other types of board;) Maybe this should be a part of widget board component.

Reply:
We have discussed this matter with Marek and I agree that this is a very valid concern. As we established, the best possible solution would be to create a separate component for these icons rendering. That component would accept currentBoardType as parameter and each boardType will have specified which actions are allowed on them and render them accordingly. However this will be implemented in separate task.